src/nbl/video/COpenGL_Queue.h

```diff
@@ -0,0 +1,161 @@
+#include "nbl/video/CEGL.h"
+#include "nbl/video/IOpenGL_FunctionTable.h"
+
+namespace nbl {
+namespace video
+{
+
+template <typename FunctionTableType_>
+class COpenGL_Queue final : public IGPUQueue
+{
+    public:
+        using FunctionTableType = FunctionTableType_;
+        using FeaturesType = typename FunctionTableType::features_t;
+
+    private:
+        using ThreadHandlerInternalStateType = FunctionTableType;
+        using QueueElementType = SSubmitInfo; // TODO might want to change to array of SSubmitInfo + fence
+
+        struct CThreadHandler final : public system::IAsyncQueueDispatcher<QueueElementType, ThreadHandlerInternalStateType>
+        {
+        public:
+            CThreadHandler(const egl::CEGL* _egl, FeaturesType* _features, EGLContext _master, EGLConfig _config, EGLint _major, EGLint _minor) :
+                egl(_egl),
+                masterCtx(_master), config(_config),
+                major(_major), minor(_minor),
+                thisCtx(EGL_NO_CONTEXT), pbuffer(EGL_NO_SURFACE),
+                features(_features)
+            {
+
+            }
+
+        protected:
+            using base_t = system::IAsyncQueueDispatcher<QueueElementType, ThreadHandlerInternalStateType>;
+
+            ThreadHandlerInternalStateType init() override
+            {
+                egl->call.peglBindAPI(EGL_OPENGL_API);
+
+                const EGLint ctx_attributes[] = {
+                    EGL_CONTEXT_MAJOR_VERSION, major,
+                    EGL_CONTEXT_MINOR_VERSION, minor,
+                    EGL_CONTEXT_OPENGL_PROFILE_MASK, EGL_CONTEXT_OPENGL_CORE_PROFILE_BIT,
+
+                    EGL_NONE
+                };
+
+                thisCtx = egl->call.peglCreateContext(egl->display, config, masterCtx, ctx_attributes);
+
+                // why not 1x1?
+                const EGLint pbuffer_attributes[] = {
+                    EGL_WIDTH, 128,
+                    EGL_HEIGHT, 128,
+
+                    EGL_NONE
+                };
+                pbuffer = egl->call.peglCreatePbufferSurface(egl->display, config, pbuffer_attributes);
+
+                egl->call.peglMakeCurrent(egl->display, pbuffer, pbuffer, thisCtx);
+
+                return ThreadHandlerInternalStateType(&egl->call, features);
+            }
+
+            void processElement(ThreadHandlerInternalStateType& _gl, queue_element_t&& e) const override
+            {
+                static_assert(std::is_same_v<ThreadHandlerInternalStateType, FunctionTableType>);
+                // a cast to common base so that intellisense knows function set (can and should be removed after code gets written)
+                IOpenGL_FunctionTable& gl = static_cast<IOpenGL_FunctionTable&>(_gl);
+                // wait semaphores
+                // TODO glMemoryBarrier() corresponding to _submit.pWaitDstStageMask[i]
+                // submit commands to GPU
+                // glFlush?
+                // signal semaphores
+                // glFinish
```

Comment:
I dont get this.

Reply:
what exactly?

Reply:
making a note for myself, I havent got the time to familiarize myself with the code of this PR as w hole yet.